airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -502,3 +514,152 @@ def test_connection(self):
             return success
         except Exception as e:
             return False, str(e)
+
+
+class WasbAsyncHook(WasbHook):
+    """
+    An async hook that connects to Azure WASB to perform operations.
+
+    :param wasb_conn_id: reference to the :ref:`wasb connection <howto/connection:wasb>`
+    :param public_read: whether an anonymous public read access should be used. default is False
+    """
+
+    def __init__(
+        self,
+        wasb_conn_id: str = "wasb_default",
+        public_read: bool = False,
+    ) -> None:
+        """Initialize the hook instance."""
+        self.conn_id = wasb_conn_id
+        self.public_read = public_read
+        self.blob_service_client: AsyncBlobServiceClient = self.get_conn()
+
+    def get_conn(self) -> AsyncBlobServiceClient:
+        """Return the Async BlobServiceClient object."""
+        conn = self.get_connection(self.conn_id)
```

Comment:
Does this need to be wrapped in a `sync_to_async()`?

Reply:
Done